pkg/archive/archive_windows.go

```diff
@@ -6,9 +6,25 @@ import (
 	"archive/tar"
 	"fmt"
 	"os"
+	"path/filepath"
 	"strings"
 )
 
+// fixVolumePathPrefix does platform specific processing to ensure that if
+// the path being passed in is not in a volume path format, convert it to one.
+func fixVolumePathPrefix(srcPath string) string {
+	if !strings.HasPrefix(srcPath, `\\?\`) {
+		srcPath = `\\?\` + srcPath
+	}
+	return srcPath
+}
+
+// getWalkRoot calculates the root path when performing a TarWithOptions.
+// We use a seperate function as this is platform specific.
+func getWalkRoot(srcPath string, include string) string {
+	return filepath.Join(srcPath, include)
```

Comment:
On Windows, using the long file path semantics, you are guaranteed that the path will never end with a . as those semantics don't support it. Hence the "hack" on Linux is not necessary. The filepath.Join here is cleaner.